### ISSUE TITLE
generateLibrariesJson doesn't respect dependency classifier/extension

### ISSUE THREAD

**Issue by Galster-dev** [Minecrell/plugin-yml#44]
Exported dependency coords do not include classifiers and extensions. This leads to some bugs, e.g. server trying to download `jar` artifact for `pom` dependency (with `PluginLoader` implementation presented in README)

My personal workaround atm is exporting artifacts instead of dependencies, but you may not like it because of increased JSON file size.

**Comment by stephan-gh** [comment]
Could we just filter out pom dependencies completely?

**Comment by Galster-dev** [comment]
Some pom dependencies are meta packages containing a few jar artifacts

**Comment by Galster-dev** [comment]
My actual issue happened with [InvUi](https://github.com/NichtStudioCode/InvUI/) (pom available [here](https://repo.xenondevs.xyz/releases/xyz/xenondevs/invui/invui/1.14/invui-1.14.pom)). This dependency unfolds to `inventory-access-{r1-r14}` and `invui-core` which are jar artifacts and required at runtime.

**Comment by stephan-gh** [comment]
How would you expect to handle these on the PluginLoader side? Does it provide the necessary functionality to parse & handle metadata packages?

**Comment by Galster-dev** [comment]
Yes, `DefaultArtifact` has both constructors with classifiers/extension and can parse it from a single so called "coords" string (format available [here](https://maven.apache.org/resolver/apidocs/org/eclipse/aether/artifact/DefaultArtifact.html#%3Cinit%3E(java.lang.String)))
